eth/handler.go

```diff
@@ -399,3 +410,28 @@ func (h *handler) txBroadcastLoop() {
 		}
 	}
 }
+
+// missingBodyLoop listens to the MissingBody event in Slice and calls the blockAnnounces.
+func (h *handler) missingBodyLoop() {
+	defer h.wg.Done()
+	for {
+		select {
+		case header := <-h.missingBodyCh:
+			// Check if any of the peers have the body
+			for _, peer := range h.peers.peers {
+				if peer.KnownBlock(header.Hash()) {
+					log.Debug("Fetching the missing body from", "peer", peer.ID(), "hash", header.Hash(), "number", header.NumberU64())
+					h.blockFetcher.Notify(peer.ID(), header.Hash(), header.NumberU64(), time.Now(), peer.RequestOneHeader, peer.RequestBodies)
```

Comment:
Why are you fetching the header? We already have the header.

Reply:
block_fetcher is built to handle block hash broadcasts. This requires it to get the header and body and call insert chain. Since we are using the current infrastructure of block_fetcher to get the missing body, there doesn't seem to be an easy fix. We can come back in the future to optimize or find a better solution to it.